### PR TITLE
Upgrade to OkHttp 4

### PR DESCRIPTION
**Pull request by CyborgRahil**
This PR is my proposal for #741 
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description
Upgrading to OkHttp 4

## :pencil: Checklist

- [x] I ran `./gradlew spotlessApply` before submitting the PR
- [x] I reviewed submitted code
- [ ] I added tests to verify changes
- [x] All tests passing

### CODE REVIEW

File: core/src/test/java/io/plaidapp/core/dribbble/data/TestData.kt [hunk 1]
```diff
@@ -57,4 +57,4 @@ val shots = listOf(
     )
 )
 
-val errorResponseBody = ResponseBody.create(MediaType.parse(""), "Error")
+val errorResponseBody = "Error".toResponseBody("".toMediaTypeOrNull())
```
Comment (ashdavies) [inline core/src/test/java/io/plaidapp/core/dribbble/data/TestData.kt]: I know this is part of OkHttp, but extension functions converting “primitive” types to specific types feel all kinds of wrong

Comment (CyborgRahil) [reply core/src/test/java/io/plaidapp/core/dribbble/data/TestData.kt]: > I know this is part of OkHttp, but extension functions converting “primitive” types to specific types feel all kinds of wrong

I understand but If you go to the implementation of create() internally it does the same thing 
```fun create(contentType: MediaType?, content: String) = content.toResponseBody(contentType)```
This part is only used as data for test.

Comment (emartynov) [reply core/src/test/java/io/plaidapp/core/dribbble/data/TestData.kt]: Sorry to jump to the discussion. Calling the method with two params is more readable than a combined function from two extension functions.

Comment (keyboardsurfer) [reply core/src/test/java/io/plaidapp/core/dribbble/data/TestData.kt]:
```suggestion
val errorResponseBody = ResponseBody.create(MediaType.parse(""), "Error")
```

Comment (CyborgRahil) [reply core/src/test/java/io/plaidapp/core/dribbble/data/TestData.kt]: @keyboardsurfer 
As per documentation of Square (upgrading_to_okhttp_4): [DOC link](https://square.github.io/okhttp/upgrading_to_okhttp_4/)
-  create(contentType: MediaType?, content: String) is deprecated .Moved to extension function. Put the 'content' argument first. 
- Using 'parse(String): MediaType?' is an error. We need to moved to extension function.

Thats why I was replaced ResponseBody.create(MediaType.parse(""), "Error") to  
"Error".toResponseBody("".toMediaTypeOrNull()).

After reverting this due to your change request It cause CI error:
`e: /home/circleci/plaid/core/src/test/java/io/plaidapp/core/producthunt/data/TestData.kt: (25, 55): Using 'parse(String): MediaType?' is an error. moved to extension function`

> _Which is expected as per okhttp doc ._

If you all are agree with above points can i again revert to my previous change?

Comment (keyboardsurfer) [reply core/src/test/java/io/plaidapp/core/dribbble/data/TestData.kt]: Sounds good, please go ahead.

**Comment by CyborgRahil** [comment]
@keyboardsurfer 
CI failure but locally every test cases works fine. I did not have access. Can you please retrigger it?

**Pull request merged**